backend/tests/services/test_context_service.py

```
@@ -18,20 +18,24 @@ class TestContextServiceUpload:
-        """Test upload fails for unsupported file types"""
-        from app.services.context import context_service
+        """Test upload fails for binary files with unknown extensions via MIME detection"""
+        from app.services.attachment.parser import DocumentParseError, DocumentParser
 
         # Arrange
-        mock_db = Mock()
-        user_id = 1
-        filename = "test.exe"
-        binary_data = b"fake exe content"
+        parser = DocumentParser()
+        filename = "test.bin"
```

Ruff (0.14.10)

Local variable `filename` is assigned to but never used

Remove assignment to unused variable `filename`

(F841)